### ISSUE TITLE
Keepass autotype failed with @ and | caracteres in password 

### ISSUE THREAD

**Issue by damienDWD** [jimradford/superputty#544]
Hello,
I uses Keepass to store all my credentials internet&console and use the autotype keepass functionnality (Ctr+Alt+A shortcut) to loggin on putty session however, I have an issue when I have special caracteres like @ and | in one field (login/password).

I have open an issue on the Keepass sourceforge project (https://sourceforge.net/p/keepass/bugs/1430/) but like the problem is not reproductible on the classic putty and appears only with SuperPutty (last version 1.4.0.6), I think that you could investigate too. Please find a screen capture here: https://sourceforge.net/p/keepass/bugs/_discuss/thread/0e4d35c1/f3ee/attachment/Bug%20Autotype.avi

Best regards,
Damien LEVEQUE

**Comment by jimradford** [comment]
@damienDWD in your video, it looks like your password is being sent as the username (with the @ symbol in it). Is that your password or is that your username?

**Comment by damienDWD** [comment]
Hello and thank you for your answer.
Like the password was declined during the SSH connexion, I has tries to change the autotype sequence to understand the problem. Please find in my keepass hyperlink the new informations like a new video screenshot with the username/password "h=|d@k#a^b{c}d[e]f"

Regards,
Damien LEVEQUE

**Comment by damienDWD** [comment]
Hello,
Have you any news ? I try to install the new version of superputty 1.4.0.7 but there is no difference, same bug.

Regards,
Damien LEVEQUE

**Comment by damienDWD** [comment]
Hello,
I obtained a workaround from the keepass support, the problem comes from the keyboard language.
If I change my language (in the taskbar) by English, it's work fine. Have you a better solution to support other keyboards like French or French (Belgium) ?

Best regards,
Damien LEVEQUE

**Comment by damienDWD** [comment]
Hello,
I obtained a workaround from the keepass support, the problem comes from the keyboard language.
If I change my language (in the taskbar) by English, it's work fine. Have you a better solution to support other keyboards like French or French (Belgium) ?

Best regards,
Damien LEVEQUE

**Comment by jimradford** [comment]
@damienDWD Thanks for the report. We just pass through the data to putty, so I am not sure where we could implement any sorta fix without knowing more what is happening with keepass. I'll follow up with the keepass support issue in a couple weeks.

**Comment by damienDWD** [comment]
Hello,
Could you please see the last informations on the keepass case (https://sourceforge.net/p/keepass/bugs/1430/) ?

It doesn't provide any solution in order to the autotype works with other software like putty / notepad / firefox / etc...

Regards,
Damien LEVEQUE

**Comment by jimradford** [comment]
@damienDWD take a look at issue #457 and see if the workaround provided in that issue helps at all.

**Comment by damienDWD** [comment]
Hello and thank you for your answer but the workaround doesn't work.
In other I have obtain an explication and an workaround from on of the keepass developpers (Dominik Reichl):
-------------------------------------------------------------------------------------------------------------------------------
First of all, PuTTY's default Ctrl+Alt behavior differs from Windows' behavior; KeePass implements a workaround for this.

SuperPuTTY and MTPuTTY spawn PuTTY processes whose windows are displayed embedded in the SuperPuTTY/MTPuTTY window (without window borders). As the PuTTY window in this case doesn't have usual main window properties, various window enumeration methods fail to find this window (especially the method of the .NET Framework fails, and KeePass used this one), thus KeePass didn't activate the workaround.

I've now implemented a new target window classification method, which improves compatibility with target windows hosted within other windows (e.g. a PuTTY window within SuperPuTTY/MTPuTTY).

In short, it should work fine now with SuperPuTTY and MTPuTTY.

Here's the latest development snapshot for testing:
http://keepass.info/filepool/KeePass_151019b.zip
-------------------------------------------------------------------------------------------------------------------------------

**Comment by jimradford** [comment]
Thanks @damienDWD for the update, and send my regards to Dominik for his efforts.
